tests/syslog/test_syslog_counter.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import pytest
+
+from tests.common.config_reload import config_reload
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology("any")
+]
+
+
+@pytest.fixture(autouse=True, scope="module")
+def enable_syslog_counter(rand_selected_dut):
+    # Older version image may not support syslog counter feature
+    yang = rand_selected_dut.command("sudo cat /usr/local/yang-models/sonic-device_metadata.yang")['stdout']
+    if "syslog_counter" not in yang:
+        pytest.skip("syslog_counter feature is not supported in this image.")
+        return
+
+    rand_selected_dut.command('sudo sonic-db-cli CONFIG_DB hset "DEVICE_METADATA|localhost" "syslog_counter" "true"')
+    rand_selected_dut.command('sudo config save -y')
+    config_reload(rand_selected_dut, safe_reload=True)
```

Comment:
Is a config reload mandatory to make this configuration change work?

Reply:
Another question, what about enable this syslog_counter feature by default?

Reply:
Yes, we need reload config to re-render the rsyslog config file.
The syslog_counter will use ~8MB memory and 5% CPU, so we want to disable it by default to save resource.